File: 백준/Silver/15565. 귀여운 라이언/귀여운 라이언.py
n,k = map(int,input().split())
dolls = list(map(int,input().split()))

start = 0
end = 0
min_len = int(1e9)
cnt = 0

while True:
    if cnt >= k:
        min_len = min(min_len, end - start)
        if dolls[start] == 1:
            cnt -= 1
        start += 1
    elif end == n:
        break
    else:
        if dolls[end] == 1:
            cnt += 1
        end += 1

print(min_len if min_len != int(1e9) else -1)

File: 백준/Silver/15565. 귀여운 라이언/귀여운 라이언.js
const input = require("fs").readFileSync("/dev/stdin", "utf-8").split("\n");

const [n, k] = input[0].trim().split(" ").map(Number);
const dolls = input[1].trim().split(" ").map(Number);

let start = 0;
let end = 0;
let minLen = Infinity;
let cnt = 0;

while (true) {
	if (cnt >= k) {
		minLen = Math.min(minLen, end - start);
		if (dolls[start] === 1) {
			cnt--;
		}
		start++;
	} else if (end === n) {
		break;
	} else {
		if (dolls[end] === 1) {
			cnt++;
		}
		end++;
	}
}

if (minLen === Infinity) {
	console.log(-1);
} else {
	console.log(minLen);
}
